sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -289,19 +408,29 @@ public void dispose() {
         tokenManager.close();
     }
 
-    Mono<Void> send(byte[] bytes, int arrayOffset, int messageFormat) {
+    Mono<Void> sendTransaction(byte[] bytes, int arrayOffset, int messageFormat, AmqpTransaction transactionId) {
+
+        Mono<DeliveryState> workToSend = Mono.create(sink -> sendTransaction(new RetriableWorkItem(bytes,
+            arrayOffset, messageFormat, sink, timeout, transactionId)));
+
         if (hasConnected.get()) {
-            return Mono.create(sink -> send(new RetriableWorkItem(bytes, arrayOffset, messageFormat, sink, timeout)));
+            return workToSend.then();
         } else {
             return RetryUtil.withRetry(
                 handler.getEndpointStates().takeUntil(state -> state == EndpointState.ACTIVE),
                 timeout, retry)
-                .then(Mono.create(sink ->
-                    send(new RetriableWorkItem(bytes, arrayOffset, messageFormat, sink, timeout))));
+                .then(workToSend)
+                .then();
         }
     }
 
-    private void send(RetriableWorkItem workItem) {
+    /**
+     * Add the work item in pending send to be processed on {@link ReactorDispatcher} thread.
+     *
+     * @param workItem to be processed.
+     */
+    void sendTransaction(RetriableWorkItem workItem) {
```

Comment:
This isn't always the case that there is a transaction with the send. Method name should remain as send.

Also, why is it package-private?